### PR TITLE
add create post page

### PR DESCRIPTION
**Pull request by AnnaViktorovna**
You can create a post only if you have authorized the user. After clicking the "Publish" button, the user is redirected to the posts page. The user can add one picture, delete a picture or change the picture, but this is not necessary. But the title and description are mandatory
Screen Shot 2024-12-21 at 3 45 18 PM
Screen Shot 2024-12-21 at 3 45 39 PM

### CODE REVIEW

**Review by tianahR — COMMENTED** [review]
Hi Anna ! title and content are required field in the Post Model. Image is not. Only authenticated user can create post. In your handlePost, I think you should call the backend api endoint "createPost", using axios. 

Something like this 

....
const response = await axios.post('/api/v1/posts', formData, {
                headers: {
                    'Content-Type': 'multipart/form-data',
                    Authorization: `Bearer ${localStorage.getItem('token')}`, // Assuming JWT
                },
            });

File: src/App.jsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ function App() {
           <Route path="/" element={<Home />} />
           <Route path="/login" element={<Login />} /> 
           <Route path="/register" element={<Register />} /> 
-          <Route path="/posts" element={<PostsPage />} />
+          <Route path="/posts/" element={<PostsPage />} />
```
Comment (vitaliipp) [inline src/App.jsx]: Please don't add slash at the end
should be <Route path="/posts" element={<PostsPage />} />

**Review by vitaliipp — CHANGES_REQUESTED** [review]
Please add a screenshot and description, and ensure there is a margin at the top
image

File: src/App.jsx (hunk 1 above)
Comment (AnnaViktorovna) [reply src/App.jsx]: ok. Thanks

**Review by tianahR — APPROVED** [review]
Approved

**Review by vitaliipp — APPROVED** [review]
Good job 🚀

**Pull request merged**